src/Mvc/Collection.php

```diff
@@ -0,0 +1,1555 @@
+<?php
+
+/** @noinspection PhpUnused */
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalconphp.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Phalcon\Incubator\Mvc;
+
+use JsonSerializable;
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\Serializable as BsonSerializable;
+use MongoDB\BSON\Unserializable;
+use MongoDB\Database;
+use MongoDB\Driver\Cursor;
+use Phalcon\Di;
+use Phalcon\Di\AbstractInjectionAware;
+use Phalcon\Di\DiInterface;
+use Phalcon\Events\ManagerInterface as EventsManagerInterface;
+use Phalcon\Helper\Str;
+use Phalcon\Incubator\Mvc\Collection\BehaviorInterface;
+use Phalcon\Incubator\Mvc\Collection\Exception;
+use Phalcon\Incubator\Mvc\Collection\ManagerInterface;
+use Phalcon\Messages\Message;
+use Phalcon\Messages\MessageInterface;
+use Phalcon\Mvc\EntityInterface;
+use Phalcon\Validation\ValidationInterface;
+use Serializable;
+
+/**
+ * Class Collection
+ *
+ * ActiveRecord class for the management of MongoDB collections
+ *
+ * @package Phalconator\Mvc
+ */
+class Collection extends AbstractInjectionAware implements
+    CollectionInterface,
+    EntityInterface,
+    Serializable,
+    BsonSerializable,
+    Unserializable,
+    JsonSerializable
+{
+    public const DIRTY_STATE_PERSISTENT = 0;
+    public const DIRTY_STATE_TRANSIENT = 1;
+    public const DIRTY_STATE_DETACHED = 2;
+
+    public const OP_NONE = 0;
+    public const OP_CREATE = 1;
+    public const OP_UPDATE = 2;
+    public const OP_DELETE = 3;
+
+    /**
+     * @var ObjectId|mixed $_id
+     */
+    protected $_id;
```

Comment:
Is it possible to remove `_` from variable name?

Reply:
No, MongoDb imposes a _id field in all the documents of a collection. The old Phalcon MDO also used this syntax

Reply:
https://docs.mongodb.com/manual/core/document/#field-names

Reply:
https://docs.mongodb.com/manual/core/document/#the-id-field

Reply:
Then we need to exclude that for PHP CS.

Reply:
Yeah, I haven't figured out how to exclude just that field yet.